### DISCUSSION TITLE
[Question]: Does LibreChat support connecting an AWS Bedrock Agent?

### DISCUSSION THREAD

**Discussion by oliveunji** [danny-avila/LibreChat#4505] [Q&A]
### What is your question?

Can I integrate AWS Bedrock Agent within LibreChat? I'm wondering if this functionality is already supported or not. 

### More Details

Hi team, 
I have already checked that LibreChat can be integrated with an AWS Bedrock account to use the models provided by Bedrock. Thanks for the support!
We have also created a custom chatbot for our company using AWS Bedrock's [KnowledgeBase](https://aws.amazon.com/bedrock/knowledge-bases/) and [Agent](https://aws.amazon.com/bedrock/agents/?nc1=h_ls) features, and we'd like to use this within LibreChat. I'm wondering if this functionality is already supported?

### What is the main subject of your question?

Endpoints

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Bedrock agents are not supported and will not be supported. However, our own "agents" implementation will soon release, for which you can use Bedrock models.


AWS Bedrock's [KnowledgeBase](https://aws.amazon.com/bedrock/knowledge-bases/) is a different story and will be supported.

**Comment by gomesdigital** [comment]
Hi @danny-avila, 

Is support for knowledge bases actively being worked on? How certain is it that it will become a feature?

**Comment by ala85** [comment]
Is there an example of librechat configuration on how to use agents with Bedrock?

**Reply by danny-avila** [reply]
It's simple, just follow this: https://www.librechat.ai/docs/features/agents

agents need to be enabled if using `ENDPOINTS` environment variable, i.e. `ENDPOINTS=agents,#...rest`

![chrome_LcNUTX1fNM](https://github.com/user-attachments/assets/a4177186-ee57-4f7d-a8cc-041231ea3bc1)

**Reply by ala85** [reply]
I previously looked at the page but there is no mention of agents configuration in there, nor on librechat.yaml configuration.

can an example be shared? I have defined endpoints but cannot see any documented agents property under it

**Reply by danny-avila** [reply]
`ENDPOINTS=agents` along with any other values you have set, i.e `ENDPOINTS=agents,bedrock`

There is no YAML configuration other than disabling the feature completely:

https://www.librechat.ai/docs/configuration/librechat_yaml/object_structure/interface#agents

**Reply by rromanchuk** [reply]
@danny-avila  I think @ala85 means

```
POST /agents/agentId/agentAliases/agentAliasId/sessions/sessionId/text HTTP/1.1
```

Which uses multi-agent collaborators tightly coupled to foundation model and bedrock kb's
